tests/storages_tests/test_storages.py

```diff
@@ -352,336 +452,406 @@ def test_get_trial_number_from_id(storage_mode):
-        datetime_before = datetime.now()
+        # Cannot set attributes of finished trials.
+        storage.set_trial_state(trial_id_1, TrialState.COMPLETE)
+        with pytest.raises(RuntimeError):
+            storage.set_trial_system_attr(trial_id_1, "key", "value")
 
-        trial_id = _create_new_trial_with_example_trial(
-            storage, study_id, EXAMPLE_DISTRIBUTIONS, example_trial
-        )
 
-        datetime_after = datetime.now()
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_get_all_study_summaries(storage_mode: str) -> None:
 
-        trial = storage.get_trial(trial_id)
-        _check_example_trial_static_attributes(trial, example_trial)
-        if trial.state.is_finished():
-            assert trial.datetime_start is not None
-            assert trial.datetime_complete is not None
-            assert datetime_before < trial.datetime_start < datetime_after
-            assert datetime_before < trial.datetime_complete < datetime_after
-        else:
-            assert trial.datetime_start is not None
-            assert trial.datetime_complete is None
-            assert datetime_before < trial.datetime_start < datetime_after
+    with StorageSupplier(storage_mode) as storage:
+        expected_summaries, _ = _setup_studies(storage, n_study=10, n_trial=10, seed=46)
+        summaries = storage.get_all_study_summaries()
+        assert len(summaries) == len(expected_summaries)
+        for study_id, expected_summary in expected_summaries.items():
+            summary = None  # type: Optional[StudySummary]
+            for s in summaries:
+                if s.study_name == expected_summary.study_name:
+                    summary = s
+                    break
+            assert summary is not None
+            assert summary.direction == expected_summary.direction
+            assert summary.datetime_start == expected_summary.datetime_start
+            assert summary.study_name == expected_summary.study_name
+            assert summary.n_trials == expected_summary.n_trials
+            assert summary.user_attrs == expected_summary.user_attrs
+            assert summary.system_attrs == expected_summary.system_attrs
+            if expected_summary.best_trial is not None:
+                assert summary.best_trial is not None
+                _check_trial_equality(summary.best_trial, expected_summary.best_trial)
 
 
-@parametrize_storage
-def test_get_all_trials(storage_init_func):
-    # type: (Callable[[], BaseStorage]) -> None
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_get_trial(storage_mode: str) -> None:
 
-    storage = storage_init_func()
-    study_id_1 = storage.create_new_study()
-    study_id_2 = storage.create_new_study()
+    with StorageSupplier(storage_mode) as storage:
+        _, study_to_trials = _setup_studies(storage, n_study=2, n_trial=20, seed=47)
 
-    datetime_before = datetime.now()
+        for study_id, expected_trials in study_to_trials.items():
```

Comment:
`study_id` is not accessed in the scope.

```suggestion
        for _, expected_trials in study_to_trials.items():
```